### PR TITLE
Ignore strict errors by `@babel/parser`

### LINKED ISSUE #11753: SyntaxError: Assigning to 'arguments' in strict mode.
**Environments:**

- Prettier Version: 2.4.1
- Running Prettier via: CLI
- Runtime: node 12.22.7
- Operating System: Linux
- Prettier plugins (if any): None

**Steps to reproduce:**

```
echo "function myfunc() { arguments = arguments }" | yarn -s prettier --stdin-filepath script.js
```

**Expected behavior:**

A nicely formatted output of my function.

**Actual behavior:**

```
[error] script.js: SyntaxError: Assigning to 'arguments' in strict mode. (1:21)
[error] > 1 | function myfunc() { arguments = arguments }
[error]     |                     ^
[error]   2 |
```

### LINKED ISSUE #11754: SyntaxError: Binding 'arguments' in strict mode.
**Environments:**

- Prettier Version: 2.4.1
- Running Prettier via: CLI
- Runtime: node 12.22.7
- Operating System: Linux
- Prettier plugins (if any): None

**Steps to reproduce:**

```
echo "function myfunc() { var arguments }" | yarn -s prettier --stdin-filepath script.js
```

**Expected behavior:**

A nicely formatted output of my function.

**Actual behavior:**

```
[error] script.js: SyntaxError: Binding 'arguments' in strict mode. (1:25)
[error] > 1 | function myfunc() { var arguments }
[error]     |                         ^
[error]   2 |
```

### LINKED ISSUE #11755: SyntaxError: In strict mode code, functions can only be declared at top level or inside a block.
**Environments:**

- Prettier Version: 2.4.1
- Running Prettier via: CLI
- Runtime: node 12.22.7
- Operating System: Linux
- Prettier plugins (if any): None

**Steps to reproduce:**

```
echo "foo: function bar() {}" | yarn -s prettier --stdin-filepath script.js
```

**Expected behavior:**

A nicely formatted output of my function.

**Actual behavior:**

```
[error] script.js: SyntaxError: In strict mode code, functions can only be declared at top level or inside a block. (1:6)
[error] > 1 | foo: function bar() {}
[error]     |      ^
[error]   2 |
```

### LINKED ISSUE #11756: SyntaxError: Binding 'eval' in strict mode.
**Environments:**

- Prettier Version: 2.4.1
- Running Prettier via: CLI
- Runtime: node 12.22.7
- Operating System: Linux
- Prettier plugins (if any): None

**Steps to reproduce:**

```
echo "var eval" | yarn -s prettier --stdin-filepath script.js
```

**Expected behavior:**

A nicely formatted output of my function.

**Actual behavior:**

```
[error] script.js: SyntaxError: Binding 'eval' in strict mode. (1:5)
[error] > 1 | var eval
[error]     |     ^
[error]   2 |

```

### LINKED ISSUE #11758: SyntaxError: Assigning to 'eval' in strict mode.
**Environments:**

- Prettier Version: 2.4.1
- Running Prettier via: CLI
- Runtime: node 12.22.7
- Operating System: Linux
- Prettier plugins (if any): None

**Steps to reproduce:**

```
echo "eval = 1" | yarn -s prettier --stdin-filepath script.js
```

**Expected behavior:**

A nicely formatted output of my function.

**Actual behavior:**

```
[error] script.js: SyntaxError: Assigning to 'eval' in strict mode. (1:1)
[error] > 1 | eval = 1
[error]     | ^
[error]   2 |
```

### LINKED ISSUE #11777: Unstable formatting on multi-line comment after type
**Prettier 2.4.1**
- [Playground link (1st format)](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAEjGDAWThgAsIATAHSgHoAqB27B7AFTJwAMBhAeQByggKK923bAFtSFStjgBnGAEMARgBsAloq6LsK7DACuUKHA1GIRrtkVwATgDdH2LZQQwtAMy1x5amg2cCxsqg4A5qTYDkoQxg5gcAB0oXTYALws2AA+2NQgAsJi7AW02YzMUNg1bJw8ACIiADIi7CKSMuRU2B4apErBdlhgPn7ysYrxiSnZrHTZeQVNre1lUBVMc3W23ADibZ2yPbEAjsZKMPqGsRiTnipe0NgQ3kOKI2P+MXEJSanVWrYABKcHOl30xkUWigEWwB3Ydgoxg08mgGiCsRgDj8Ll6jxUAJq80W+RACIKAG4QAAaEAQDBPKCKZCgFQOBwQADuAAV2QgWSgVBouSo0Cy6WoHCowABrUgAZRUMmaMLgyG8wvskulcsVGBlMIiyGxFzpcCkan8HkozRUsOMKiiADEIA4pI8vLDkCAVMYYBBaSAyDApBoAOpkLTwD4yuAKgXRrROaNoH1gRQSkAw+wOGA86URD0arVwOkAK0UAA8FUb+gBFYwQeAljTakAGhy5n3oLCKMA4xlBu4wmDh9zkZAADgADHS7hB7OHpRgfXclI4XEHzs24AWGYLfYoALTmfz+INnYxaWIFp3FpCattlkD2KRaE0OM2vutwRu71t21UNRx0oSckAAJjpbEVC0bRYV4CApAfEAlAAViDKE4HYdRBSfdsnAuABJKAPFgBUBy0RkAEFSIVdB+kAuAAF9mKAA)
- [Playground link (2nd format)](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAEjGDAWThgAsIATbAXmwHoAqRgHSm0ewBUycADAMIB5AHIiAogK59sAW1IVqcAM4wAhgCMANgEtlvZdjXYYAVyhQ4WkxBO9syuACcAbs+w7KCGDoBmOuGoNNDs4Ng4TNScAc1JsJxUIUycwOAA6cMZ6cOwAH2wWEGExSS5CthymVnZsCJ5+ABFxABlxLnEZeXIqbC8tUhVQhywwPwDqBOUklPScjmya-MKm1vbyqErmOc567D4AcTbOhR6EgEdTFRhDYwSMSe81H2hsCF8h5RGxwPjE5NSMjUIgAlOAXK6GUzKHRQaLYQ5cBwUUxaajQLQhBIwJwBNy9J5qQG1eY5JYgBGFADcbBAABoQBAMM8oMpkKAok4IAB3AAKUQQrJQai0XLUaFZ9I0TjUYAA1qQAMpqeTNGFwZC+YWOSXSuWKjAymHRZDYy70uCyDSBLyUZpqWGmNSxABiECcsiePlhyBAalMMAgdJAZBgsi0AHUyDp4J8ZXAFQLozoXNG0D6wMoJSAYY4nDAedLoh6NVq4PSAFbKAAeCqN-QAiqYIPAS1ptSADU5cz70FhlGAcUyg-cYTBw55yMgABwABnp9wgjnD0owPvuKmcbiDF2bcALjMFvuUAFpLIFAkHzqYdAkC07i0hNW2yyBHLIdCanGbX3W4I3d627bqBo46UJOSAAEz0tiag6LosICBAsgPiAKgAKxBlCcBcJogpPu2LiXAAklAXiwAqA46EyACCpEKug-SAXAAC+zFAA)

```sh
# Options (if any):
--parser typescript
```

`typescript` is the default for `.ts` files with the VS Code extension: https://github.com/prettier/prettier-vscode/blob/c75c9d3f29666d7c2fbceeee504c54c9a9e7f968/src/BrowserModuleResolver.ts#L79

**Input (1st format):**

```jsx
type HttpMethod
/**
 * The `CONNECT` method establishes a tunnel to the server identified by the
 * target resource.
 */ =
  | "CONNECT"

  /**
   * The `DELETE` method deletes the specified resource.
   */
  | "DELETE"

  /**
   * The `GET` method requests a representation of the specified resource.
   * Requests using GET should only retrieve data.
   */
  | "GET";
```

**Output (1st format):**

```jsx
type HttpMethod = /**
 * The `CONNECT` method establishes a tunnel to the server identified by the
 * target resource.
 */
  | "CONNECT"

  /**
   * The `DELETE` method deletes the specified resource.
   */
  | "DELETE"

  /**
   * The `GET` method requests a representation of the specified resource.
   * Requests using GET should only retrieve data.
   */
  | "GET";
```

**Input (2nd format):**

```jsx
type HttpMethod = /**
 * The `CONNECT` method establishes a tunnel to the server identified by the
 * target resource.
 */
  | "CONNECT"

  /**
   * The `DELETE` method deletes the specified resource.
   */
  | "DELETE"

  /**
   * The `GET` method requests a representation of the specified resource.
   * Requests using GET should only retrieve data.
   */
  | "GET";
```

**Output (2nd format):**

```jsx
type HttpMethod
/**
 * The `CONNECT` method establishes a tunnel to the server identified by the
 * target resource.
 */ =
  | "CONNECT"

  /**
   * The `DELETE` method deletes the specified resource.
   */
  | "DELETE"

  /**
   * The `GET` method requests a representation of the specified resource.
   * Requests using GET should only retrieve data.
   */
  | "GET";
```

**Expected behavior:**

Stable formatting

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #11753
Fixes #11754
Fixes #11755
Fixes #11756
Fixes #11758

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: changelog_unreleased/javascript/11750.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#### Fix errors on deleting variables (#11750 by @fisker)
+#### Ignore errors for non-strict mode features (#11750 by @fisker, #11778 by @sosukesuzuki)
```
Comment (sosukesuzuki) [inline changelog_unreleased/javascript/11750.md:1]: these features are not important so I think we don't need to add more code examples.

**Comment by fisker** [comment]
Are you sure this fixes #11777?

File: tests/format/js/sloppy-mode/function.js [hunk 2]
```diff
@@ -0,0 +1 @@
+foo: function bar() {}
```
Comment (fisker) [inline tests/format/js/sloppy-mode/function.js:1]: Let's name this file as `labeled-function-declaration.js` ?

**Comment by sosukesuzuki** [comment]
> Are you sure this fixes #11777?

Woops. No. I mistook via GitHub Autocomplete.

File: tests/format/js/sloppy-mode/function.js (hunk 2 above)
Comment (sosukesuzuki) [reply tests/format/js/sloppy-mode/function.js:1]: `labeled-function-declaration.js` sounds good

File: changelog_unreleased/javascript/11750.md (hunk 1 above)
Comment (fisker) [reply changelog_unreleased/javascript/11750.md:1]:
```suggestion
#### Ignore errors for sloppy mode syntax (#11750 by @fisker, #11778 by @sosukesuzuki)
```

**Review by fisker — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
Wooooops, sorry I missed your suggestion..

**Comment by fisker** [comment]
You didn't, I added it few secondes ago.

**Pull request merged**